lib/CMakeLists.txt

```diff
@@ -31,19 +34,19 @@ if(WITH_XKB)
 endif()
 
 add_subdirectory(xpp)
-list(APPEND libs ${XPP_LIBRARIES})
-list(APPEND dirs ${XPP_INCLUDE_DIRS})
+add_library(xpp-interface INTERFACE)
+target_link_libraries(xpp-interface INTERFACE ${XPP_LIBRARIES})
+target_include_directories(xpp-interface INTERFACE ${XPP_INCLUDE_DIRS})
+list(APPEND libs xpp-interface)
 
 # }}}
 # Library: i3ipcpp {{{
 
 if(ENABLE_I3)
   add_subdirectory(i3ipcpp)
-  list(APPEND libs ${I3IPCPP_LIBRARIES})
-  list(APPEND dirs ${I3IPCPP_INCLUDE_DIRS})
+  list(APPEND libs i3ipc++)
 endif()
 
 # }}}
 
 set(libs ${libs} PARENT_SCOPE)
-set(dirs ${dirs} PARENT_SCOPE)
```

Comment:
Why can we remove the dirs variable?

Reply:
So CMake actually supports 3 types of values as the right-hand argument of `target_link_libraries(<target> <item>)`:

1. The path to a `.a` or `.so` file
2. A CMake target
3. A linker flag (e.g.`-lm`)

A CMake target is an object with properties associated with it. These properties can be things like include directories, compile-time definitions, etc. When a target is used as the right hand argument of target_link_libraries, it passes these properties on the target on the left-hand side.

Before this change, all of the items in the `lib` list were regular `.a` or `.so` files, which is why the include directories needed to be handled separately. I changed them so that some of the libs are now CMake targets, which can contain their include directories directly.

Reply:
This makes a lot more sense